server/executor/selector_based_poller_executor.go

```diff
@@ -0,0 +1,65 @@
+package executor
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kubeshop/tracetest/server/model"
+)
+
+const (
+	selectorBasedPollerExecutorRetryHeader = "SelectorBasedPollerExecutor::retryCount"
+	selectorBasedPollerExecutorMaxTries    = 3
+)
+
+type selectorBasedPollerExecutor struct {
+	pollerExecutor PollerExecutor
+}
+
+func NewSelectorBasedPoller(innerPoller PollerExecutor) PollerExecutor {
+	return selectorBasedPollerExecutor{innerPoller}
+}
+
+func (pe selectorBasedPollerExecutor) ExecuteRequest(request *PollingRequest) (bool, string, model.Run, error) {
+	ready, reason, run, err := pe.pollerExecutor.ExecuteRequest(request)
+	if !ready {
+		request.SetHeader(selectorBasedPollerExecutorRetryHeader, "0")
+		return ready, reason, run, err
+	}
+
+	currentNumberTries := pe.getNumberTries(request)
+	if currentNumberTries >= selectorBasedPollerExecutorMaxTries {
+		return true, "not all selectors matched, but trace haven't changed in a while", run, err
```

Comment:
would the end user, at some point, have a clear undestanding of what `in a while` means? Like, is the amount of retries related to this error at some point? If not, maybe add the amount of retries/time in this message?